Feat: implement accounting page

teneeto opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
Implement a new page called "Accounting page" for the QuickBooks Online (QBO) connection project.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ [37706](https://github.com/Expensify/App/issues/37706)
PROPOSAL:[ issue Description ](https://github.com/Expensify/App/issues/37706#issue-2168065950)


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
**Preconditions**
1. Make sure you have a `collect policy` created in oldDot. Follow these [steps here](https://github.com/Expensify/App/pull/38283#issuecomment-2034225953) if you're not already enabled.
2. goto Profile => Workspaces => select a workspace.

**Continue**
3. select `More features` and toggle `Accounting` on from the `Integrate` section
4. At this point, you should see  `Accounting` added to the Left sidebar.
5. select `Accounting` and click setup for qbo.

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
**Preconditions**
1. Make sure you have a `collect policy` created in oldDot. Follow these [steps here](https://github.com/Expensify/App/pull/38283#issuecomment-2034225953) if you're not already enabled.
2. goto Profile => Workspaces => select a workspace.

**Continue**
3. select `More features` and toggle `Accounting` on from the `Integrate` section
4. At this point, you should see  `Accounting` added to the Left sidebar.
5. select `Accounting` and click setup for qbo.
### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
**Preconditions**
1. Make sure you have a `collect policy` created in oldDot. Follow these [steps here](https://github.com/Expensify/App/pull/38283#issuecomment-2034225953) if you're not already enabled.
2. goto Profile => Workspaces => select a workspace.

**Continue**
3. select `More features` and toggle `Accounting` on from the `Integrate` section
4. At this point, you should see  `Accounting` added to the Left sidebar.
5. select `Accounting` and click setup for qbo.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
![Screenshot 2024-03-14 at 11 14 23](https://github.com/Expensify/App/assets/22779249/95f3a717-fede-46c7-80ef-ddca3dc86183)
------------
![Screenshot 2024-03-14 at 13 03 01](https://github.com/Expensify/App/assets/22779249/afc82a12-fa9a-48c9-a32d-1219bdae26b5)


</details>
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

teneeto committed:
add initial english strings

teneeto committed:
add initial spanish strings

teneeto committed:
add accounting navigation param type

teneeto committed:
add accounting to workspace screens

teneeto committed:
add workspace acoounting route

teneeto committed:
add workspace accounting to workspace setting navigator

teneeto committed:
add workspace accounting to link to screens

teneeto committed:
add workspace accounting Page

teneeto committed:
add workspace accounting Page LHN navigation button

teneeto committed:
fix lint

teneeto committed:
add qbo round icon

teneeto committed:
export qbo round icon

teneeto committed:
use qbo round icon

teneeto committed:
connection list item should not be interactive

teneeto committed:
add default Avatar size for menu item icons

teneeto committed:
add xero round icon

teneeto committed:
export xero round icon

teneeto committed:
add xero english strings

teneeto committed:
add xero spanish strings

teneeto committed:
use xero round icon

teneeto committed:
fix lint

teneeto committed:
add more english and spanish strings

teneeto committed:
add menu options to setup QBO

teneeto committed:
add more english and spanish strings

teneeto committed:
break up qboConnectionMenuItems

teneeto committed:
add threeDotsMenu

teneeto committed:
cleanup and add more menu options

teneeto committed:
add more english and spanish strings

teneeto committed:
add disconnect integration confirm modal

teneeto committed:
add more english and spanish strings

teneeto committed:
add activity indiccator while importing

teneeto committed:
update accounting navigation

teneeto committed:
update integration icons structure

teneeto committed:
add more english and spanish strings

teneeto committed:
add iconRight props to PopoverMenu

teneeto committed:
add three-line bar to Header

teneeto committed:
merge conflict

teneeto committed:
add temp. fake qbo connection sync

teneeto committed:
revert accounting navigation

teneeto committed:
fix lint

teneeto committed:
fix dep warning

narefyev91 committed:
Merge branch 'main' into feat/37706/implement-accounting-page

narefyev91 committed:
resolve comments

narefyev91 committed:
fix lint

teneeto committed:
Merge branch 'main' of github.com:teneeto/Expensify into feat/37706/implement-accounting-page

teneeto committed:
Merge branch 'feat/37706/implement-accounting-page' of github.com:teneeto/Expensify into feat/37706/implement-accounting-page

teneeto committed:
fix review comments

teneeto committed:
resolve more comments

teneeto committed:
add key to other integrations

teneeto committed:
remove shouldOverlay prop

teneeto committed:
fix menuItem blank title on mobile

teneeto committed:
fix blank avatar icon on mobile

teneeto committed:
fix lint

teneeto committed:
fix menuItem large screens wwrapping

teneeto committed:
add threeDotsMenu to qboMenuItems

teneeto committed:
fix lint

teneeto committed:
add onPress to pressable items

teneeto committed:
Merge branch 'main' of github.com:teneeto/Expensify into feat/37706/implement-accounting-page

teneeto committed:
Merge branch 'main' of github.com:teneeto/Expensify into feat/37706/implement-accounting-page

teneeto committed:
add accounting to workspace settings stack navigator

teneeto committed:
remove disabled connection for Xero

teneeto committed:
add square icons for xero and qbo

teneeto committed:
export square icons for xero and qbo

teneeto committed:
use square icon for qbo options

teneeto committed:
threeDotsMenuItems should br renamed to overflowMenu

teneeto committed:
remove styles.pl2

teneeto committed:
add top margin to other integrations menu item

teneeto committed:
fix lint

teneeto committed:
Merge branch 'main' of github.com:teneeto/Expensify into feat/37706/implement-accounting-page

teneeto committed:
remove xero and qbo round icons

teneeto committed:
remove xero and qbo exports

teneeto committed:
use permissions for accounting routes

teneeto committed:
remove permissions for accounting routes

teneeto committed:
add hasAccessToAccountingFeatures

teneeto committed:
use hasAccessToAccountingFeatures and FullPageNotFoundView

teneeto committed:
add sync progress data

teneeto committed:
add policy connection sync data types

teneeto committed:
export PolicyConnectionSyncProgress

teneeto committed:
map PolicyConnectionSyncProgress in onyx collections

teneeto committed:
add QuickBooksOnline connection lib

teneeto committed:
add QuickBooksOnline connection button

teneeto committed:
use qbo connection button

teneeto committed:
fix lint

teneeto committed:
remove hasAccessToAccountingFeatures

teneeto committed:
fix review comments

teneeto committed:
remove fake url

teneeto committed:
pass in environmentURL

teneeto committed:
update accounting spanish translations

teneeto committed:
update accounting translations

teneeto committed:
add PolicyConnectionSyncStage types

teneeto committed:
use new PolicyConnectionSyncStage translation

teneeto committed:
Merge branch 'Expensify:main' into feat/37706/implement-accounting-page

s77rt commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
<img width="480" alt="android" src="https://github.com/Expensify/App/assets/16493223/1fdfdd14-38c5-4680-8d4f-c9cee7d1f39a">

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->
<img width="480" alt="mweb-chrome" src="https://github.com/Expensify/App/assets/16493223/474ab90c-1602-4c73-9d50-550ccbb804e3">

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
<img width="572" alt="ios" src="https://github.com/Expensify/App/assets/16493223/05b0b472-73c5-4763-9171-640714651c51">

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->
<img width="572" alt="mweb-safari" src="https://github.com/Expensify/App/assets/16493223/d905932f-ec21-4916-bf2b-7be1c0ab55c7">

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
<img width="1840" alt="web" src="https://github.com/Expensify/App/assets/16493223/f0446c58-e75d-446c-b110-3aa308117828">

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
<img width="1312" alt="desktop" src="https://github.com/Expensify/App/assets/16493223/96b56943-0c4e-45ee-947e-3d94c53b37f0">

</details>

s77rt commented on src/pages/workspace/accounting/qboConnectionButton/index.native.tsx:
```diff
@@ -0,0 +1,71 @@
+    const authToken = session?.authToken ?? null;
+
+    const renderLoading = () => <FullScreenLoadingIndicator />;
+
+    /**
+     * Handles in-app navigation for webview links
+     */
+    const handleNavigationStateChange = useCallback(({url}: WebViewNavigationEvent) => !!url, []);
+
```
What is this for exactly?

s77rt commented on src/pages/workspace/accounting/qboConnectionButton/index.native.tsx:
```diff
@@ -0,0 +1,71 @@
+function ConnectToQuickbooksOnlineButton({policyID, session}: ConnectToQuickbooksOnlineButtonPropsWithSession) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const webViewRef = useRef<WebView>(null);
```
Not used?

s77rt commented on src/pages/workspace/accounting/qboConnectionButton/index.native.tsx:
```diff
@@ -0,0 +1,71 @@
+    const webViewRef = useRef<WebView>(null);
+    const [isWebViewOpen, setWebViewOpen] = useState<boolean>(false);
+
+    const authToken = session?.authToken ?? null;
```
```suggestion
```
No need for the const

s77rt commented on src/pages/workspace/accounting/qboConnectionButton/index.native.tsx:
```diff
@@ -0,0 +1,71 @@
+
+    const authToken = session?.authToken ?? null;
+
+    const renderLoading = () => <FullScreenLoadingIndicator />;
```
Use `useCallback`

s77rt commented on src/pages/workspace/accounting/qboConnectionButton/index.native.tsx:
```diff
@@ -0,0 +1,71 @@
+                style={styles.justifyContentCenter}
+                small
+            />
+            {isWebViewOpen && (
+                <Modal
+                    onClose={() => setWebViewOpen(false)}
+                    fullscreen
+                    isVisible
```
Keep Modal always rendered and use isVisible prop
```js
isVisible={isWebViewOpen}
```

s77rt commented on src/ONYXKEYS.ts:
```diff
@@ -346,6 +346,8 @@ const ONYXKEYS = {
         /** This is deprecated, but needed for a migration, so we still need to include it here so that it will be initialized in Onyx.init */
         DEPRECATED_POLICY_MEMBER_LIST: 'policyMemberList_',
+
+        POLICY_CONNECTION_SYNC_PROGRESS: 'policyConnectionSyncProgress_',
```
Not used yet

s77rt commented on src/languages/en.ts:
```diff
@@ -2035,8 +2036,25 @@ export default {
-            importing: 'Importing customers',
             enterCredentials: 'Enter your credentials',
+            connections: {
+                syncStageName: ({stage}: PolicyConnectionSyncStageParams) => {
```
```suggestion
                syncStageName: (stage: PolicyConnectionSyncStage) => {
```
Pass the stage directly without an object

(same for Spanish)

s77rt commented on src/pages/workspace/accounting/WorkspaceAccountingPage.tsx:
```diff
@@ -188,11 +199,10 @@ function WorkspaceAccountingPage({policy}: WithPolicyProps) {
-                shouldShow={!hasAccess}
+                onBackButtonPress={PolicyUtils.goBackFromInvalidPolicy}
+                onLinkPress={PolicyUtils.goBackFromInvalidPolicy}
+                shouldShow={hasAccess}
```
```suggestion
                shouldShow={!hasAccess}
```

s77rt commented on src/libs/actions/connections/QuickBooksOnline.ts:
```diff
@@ -0,0 +1,12 @@
+import {getCommandURL} from '@libs/ApiUtils';
+import CONFIG from '@src/CONFIG';
+
+const getQuickBooksOnlineSetupLink = (policyID: string) => {
+    const callbackPath = `${CONFIG.EXPENSIFY.NEW_EXPENSIFY_URL}/workspace/${policyID}/accounting`;
+    const otherParams = new URLSearchParams({callbackPath, policyID}).toString();
+    const commandURL = getCommandURL({command: 'ConnectToQuickbooksOnline'});
+
+    return `${commandURL}&${otherParams}`;
+};
```
```suggestion
function getQuickBooksOnlineSetupLink(policyID: string) {
    const callbackPath = `${CONFIG.EXPENSIFY.NEW_EXPENSIFY_URL}${ROUTES.WORKSPACE_ACCOUNTING.getRoute(policyID)}`;
    const params = new URLSearchParams({callbackPath, policyID});
    const commandURL = getCommandURL({command: 'ConnectToQuickbooksOnline'});

    return commandURL + params.toString();
}
```

s77rt commented on src/libs/actions/connections/QuickBooksOnline.ts:
```diff
@@ -0,0 +1,12 @@
+    return `${commandURL}&${otherParams}`;
+};
+
+export default getQuickBooksOnlineSetupLink;
```
Either export an object that contains the `getQuickBooksOnlineSetupLink` function or change the file name to match the function name

teneeto committed:
Update src/pages/workspace/accounting/qboConnectionButton/index.tsx

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

teneeto committed:
Update src/pages/workspace/WorkspaceInitialPage.tsx

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

teneeto committed:
Update src/pages/workspace/accounting/WorkspaceAccountingPage.tsx

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

teneeto committed:
Update src/pages/workspace/accounting/WorkspaceAccountingPage.tsx

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

teneeto committed:
Update src/libs/actions/connections/QuickBooksOnline.ts

Co-authored-by: Abdelhafidh Belalia <16493223+s77rt@users.noreply.github.com>

teneeto committed:
add ConnectToQuickbooksOnlineButton types

teneeto committed:
use ConnectToQuickbooksOnlineButton types

teneeto committed:
use ConnectToQuickbooksOnlineButton types

teneeto committed:
Merge branch 'feat/37706/implement-accounting-page' of github.com:teneeto/Expensify into feat/37706/implement-accounting-page

teneeto committed:
fix lint

teneeto committed:
remove PolicyConnectionSyncStageParams

teneeto committed:
update syncStageName params

teneeto committed:
update syncStageName args

teneeto committed:
update filename

teneeto committed:
fix lint

teneeto committed:
fix tsc

hayata-suenaga commented on src/pages/workspace/WorkspaceInitialPage.tsx:
```diff
@@ -198,6 +200,17 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, policyMembers, r
         });
     }
 
+    if (policy?.areAccountingEnabled && policy?.areConnectionsEnabled && canUseAccountingIntegrations) {
```
I think you might have forgotten to remove `areAccountingEnabled` that you might have used to use accounting feature switching?

`areAccountingEnabled` is the property that you can check to see if the accounting/connections feature is enabled or not.

shawnborton commented:
Let us know when this is ready for another design review.

s77rt commented:
Is this waiting on a backend PR? Last I checked I got a 404 error when trying to connect quick books. I can retest/review in the next few hours.

teneeto commented:
@shawnborton, yes it is... the connection is not ready yet. but let me push a fake connection for test sake. Thanks. I'll let you know once i push it.

teneeto committed:
Merge branch 'main' of github.com:teneeto/Expensify into feat/37706/implement-accounting-page

teneeto committed:
fake a QBO connection sync

teneeto commented:
> Is this waiting on a backend PR? Last I checked I got a 404 error when trying to connect quick books. I can retest/review in the next few hours.

@s77rt I need a small [clarification here ](https://callstack-hq.slack.com/archives/C036QM0SLJK/p1712233537143759?thread_ts=1712062033.280849&cid=C036QM0SLJK) to complete that part. I will finish up ASAP once i get a response.

shawnborton commented:
Looks pretty good to me, would love a second set of eyes from @Expensify/design though (head to https://39652.pr-testing.expensify.com/settings/workspaces/43BC6076D3D1418B/accounting)

One weird thing - when not connected, you only see this:
![CleanShot 2024-04-05 at 15 38 35@2x](https://github.com/Expensify/App/assets/2319350/0b4a285e-645f-4b6d-980b-8dd0421b9a0d)

But as soon as you set up, the "More integrations" row shows up. I assume this is just because we're adding the building blocks for now?
![CleanShot 2024-04-05 at 15 39 30@2x](https://github.com/Expensify/App/assets/2319350/8b8b4855-4e85-4253-9a32-60661fbfd6f4)

dannymcclain commented:
Looking good to me too. Although since nothing really does anything right now, it's hard to get a full sense of it beyond just this one card. (But I assume that's probably expected at this point.)

>One weird thing - when not connected, you only see this:
>But as soon as you set up, the "More integrations" row shows up. I assume this is just because we're adding the building blocks for now?

Yeah I would assume that when no integrations are set up, we'd list all of them out (so no need for the other integrations accordion). Once you set one up, the ones you didn't pick in step 1 would be inside that accordion. At least that's my assumption.

s77rt commented:
Besides the unresolved comments code looks good 👍 Looking forward to the rest of the changes

hayata-suenaga commented:
> But as soon as you set up, the "More integrations" row shows up. I assume this is just because we're adding the building blocks for now?

This is a good point. @teneeto, because we're not adding Xero yet, probably we should hide the `Other integrations`.

dubielzyk-expensify commented:
> Yeah I would assume that when no integrations are set up, we'd list all of them out (so no need for the other integrations accordion). Once you set one up, the ones you didn't pick in step 1 would be inside that accordion. At least that's my assumption.

Exactly what @dannymcclain said. So in the instance with QBO being the only integration we don't actually show the "Other integrations". Though it's nice to have it built cause it'll come around quickly.

dubielzyk-expensify commented:
I believe the margin on this card should be `20px` and not `16px`. Even on mobile. @Expensify/design to confirm.

![CleanShot 2024-04-08 at 08 33 27@2x](https://github.com/Expensify/App/assets/145416827/302c44b4-92a6-4c99-b349-5459c7b2ba2e)

shawnborton commented:
Ah yes, I agree with that. We might need to fix that everywhere in case we missed it for other pages.

hayata-suenaga commented:
> Ah yes, I agree with that. We might need to fix that everywhere in case we missed it for other pages.

@s77rt, if you have time, is it possible that you can check other pages where we use cards to ensure that we're using the correct margin values? If you find any, we can open a new issue to fix them separately and I can assign you as a C+ in it 😄

s77rt commented:
@hayata-suenaga Looks like all the pages are using the 16px margin

hayata-suenaga commented:
s77rt, thank you so much for checking that 🙇 

@teneeto, some unit tests are failing. could you check these failing tests?

teneeto committed:
Merge branch 'main' of github.com:teneeto/Expensify into feat/37706/implement-accounting-page

teneeto committed:
add spacer and do not show other integrations when sync in progress

teneeto committed:
use AdminPolicyAccessOrNotFoundWrapper

teneeto committed:
remove are accounting enabled

teneeto commented:
- Added a missing Horizontal line spacer before `other integrations` and while connection is in sync, we'll hide `other integrations`. 
- Added `AdminPolicyAccessOrNotFoundWrapper` as NotFoundWrapper

let's check again cc @s77rt @shawnborton

shawnborton commented:
> Added a missing Horizontal line spacer before other integrations and while connection is in sync, we'll hide other integrations.

Ah sorry for the confusion - that must be from an old mockup, because we do not want that HR here anymore. We just want to use a nice space above the "More integrations" row. Can you please revert that?

shawnborton commented:
@teneeto I think this is ideally what the page would look like: 
![CleanShot 2024-04-09 at 11 07 44@2x](https://github.com/Expensify/App/assets/2319350/06c349c5-14fc-4895-97e4-4b532f669dfd)

Where we use a 20px gap between the "Other integrations" row and the content above it:
![CleanShot 2024-04-09 at 11 08 54@2x](https://github.com/Expensify/App/assets/2319350/1847ff74-f578-49fc-b616-0df573b69fdb)

teneeto commented:
Oh ok. Will revert now

teneeto committed:
remove horizontal line spacer

teneeto committed:
use AdminPolicyAccessOrNotFoundWrapper as parent component

teneeto commented:
Reverted @shawnborton

teneeto committed:
fix lint

shawnborton commented:
Sounds good, will run a test build again

shawnborton commented:
Hmm I can't seem to get the page to load in the test build. Any ideas?

teneeto commented:
Hi @shawnborton, I just checked and it works for me. Just a hack: If you have a Not Found page, then click on `Go back to home page`. Then you can proceed with other steps from there.

shawnborton commented:
I still can't get out of the "Not found" loop. Once I exit that, and then manually type in the `/accounting` route in the browser url, I get not found. Has the url changed by chance?

shawnborton commented:
Ah ignore me, I had to go to OldDot and make sure `p.policy.areConnectionsEnabled = true;`

s77rt commented on src/pages/workspace/accounting/WorkspaceAccountingPage.tsx:
```diff
@@ -193,54 +197,54 @@ function WorkspaceAccountingPage({policy}: WithPolicyProps) {
+        <AdminPolicyAccessOrNotFoundWrapper policyID={policyID}>
+            <FeatureEnabledAccessOrNotFoundWrapper
+                policyID={policyID}
+                featureName={CONST.POLICY.MORE_FEATURES.ARE_CONNECTIONS_ENABLED}
```
Same question as above ^

teneeto replied:
same explanation from @hayata-suenaga here: https://github.com/Expensify/App/pull/39652#discussion_r1552770390

s77rt commented on src/pages/workspace/WorkspaceInitialPage.tsx:
```diff
@@ -198,7 +200,7 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, policyMembers, r
     }
 
-    if (policy?.areAccountingEnabled) {
+    if (policy?.areConnectionsEnabled && canUseAccountingIntegrations) {
```
Or are we using the connection feature because it's the only section that's available in that page for now?

teneeto replied:
Yes. @hayata-suenaga, requested for removal here: https://github.com/Expensify/App/pull/39652#discussion_r1552770390

teneeto commented on src/pages/workspace/accounting/WorkspaceAccountingPage.tsx:
```diff
@@ -193,54 +197,54 @@ function WorkspaceAccountingPage({policy}: WithPolicyProps) {
-                shouldShow={hasAccess}
-                subtitleKey={isEmptyObject(policy) ? undefined : 'workspace.common.notAuthorized'}
+        <AdminPolicyAccessOrNotFoundWrapper policyID={policyID}>
+            <FeatureEnabledAccessOrNotFoundWrapper
```
ok

teneeto committed:
use PaidPolicyAccessOrNotFoundComponent

teneeto commented:
Hi @shawnborton 

> @teneeto what is the best way to test the "Other integrations" row? Does it do anything yet or is that for a follow up?

Hi @shawnborton, that will be a follow-up i think, it currently does nothing.

s77rt commented:
@teneeto Can you add `[No QA]` to the title seeing we don't have a clear testing steps

s77rt commented on src/pages/workspace/accounting/WorkspaceAccountingPage.tsx:
```diff
@@ -198,7 +198,7 @@ function WorkspaceAccountingPage({policy}: WithPolicyProps) {
 
     return (
         <AdminPolicyAccessOrNotFoundWrapper policyID={policyID}>
-            <FeatureEnabledAccessOrNotFoundWrapper
```
FeatureEnabledAccessOrNotFoundWrapper should be kept

trjExpensify commented:
>Hi @shawnborton, that will be a follow-up i think, it currently does nothing.

In a similar vein to how the disabled Xero row was removed earlier in this PR, showing the "Other integrations" portion that doesn't do anything until we add Xero doesn't make sense. So let's make sure we remove that before we release, and put it on @lakchote radar for the Xero implementation to add it in tandem.

teneeto committed:
revert: use FeatureEnabledAccessOrNotFoundWrapper instead

teneeto committed:
remove other integrations

s77rt commented on src/pages/workspace/accounting/WorkspaceAccountingPage.tsx:
```diff
@@ -198,7 +190,7 @@ function WorkspaceAccountingPage({policy}: WithPolicyProps) {
     return (
         <AdminPolicyAccessOrNotFoundWrapper policyID={policyID}>
-            <PaidPolicyAccessOrNotFoundComponent
+            <FeatureEnabledAccessOrNotFoundWrapper
```
I think there was a misunderstanding 😅 We should use both `PaidPolicyAccessOrNotFoundComponent` and `FeatureEnabledAccessOrNotFoundWrapper`

teneeto replied:
Ayye! got you, was even about asking. thanks for the clarity.

teneeto replied:
Done ✅

teneeto committed:
use FeatureEnabledAccessOrNotFoundWrapper and PaidPolicyAccessOrNotFoundComponent and fix lint

hayata-suenaga commented:
> So let's make sure we remove that before we release, and put it on @lakchote radar for the Xero implementation to add it in tandem.

@lakchote, I created an issue for this: https://github.com/Expensify/App/issues/39955

hayata-suenaga commented on src/pages/workspace/accounting/qboConnectionButton/index.native.tsx:
```diff
@@ -0,0 +1,71 @@
+    return (
+        <>
+            <Button
+                onPress={() => setWebViewOpen(true)}
```
is this going to open the WebView? Have you tested if the web view opens correctly or not?

teneeto replied:
Yes the webView opens correctly - although the link returns a 404 Page. I asked for more depth at that point: https://github.com/Expensify/App/pull/39652#issuecomment-2039640794

hayata-suenaga replied:
There is [another PR](https://github.com/Expensify/App/pull/39219) that implements the QBO authentication flow on web view. I just wanted to confirm how much you made progress on this so that I can inform the author of the other PR 😄

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

aldo-expensify commented on src/pages/workspace/accounting/WorkspaceAccountingPage.tsx:
```diff
@@ -0,0 +1,238 @@
+import type {AnchorPosition} from '@styles/index';
+import CONST from '@src/CONST';
+
+function WorkspaceAccountingPage({policy}: WithPolicyProps) {
```
It was agreed on the design doc that we would call this `PolicyAccountingPage`. According to the naming agreement on [here](https://expensify.slack.com/archives/C03TQ48KC/p1709066269962319?thread_ts=1706792648.640989&cid=C03TQ48KC), we should be using `Policy` instead of `Workspace` for referring to policies in text that is not user facing.

cc @hayata-suenaga

hayata-suenaga replied:
I can create a quick PR to fix the issue thank you for noticing it

aldo-expensify replied:
No worries, I'm taking care of it in the PR where we add Authentication

hayata-suenaga replied:
thank you!

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/8635193755) to staging by https://github.com/hayata-suenaga in version: 1.4.62-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/8635193755) to staging by https://github.com/hayata-suenaga in version: 1.4.62-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/8635193755) to staging by https://github.com/hayata-suenaga in version: 1.4.62-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/8732361917) to production by https://github.com/thienlnam in version: 1.4.62-17 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
